finagle-core/src/main/scala/com/twitter/finagle/dispatch/ServerDispatcher.scala

```diff
@@ -1,11 +1,65 @@
 package com.twitter.finagle.dispatch
 
 import com.twitter.finagle.context.{Contexts, RemoteInfo}
+import com.twitter.finagle.tracing.{Annotation, Trace, Tracer, TraceId}
 import com.twitter.finagle.transport.Transport
 import com.twitter.finagle.{Service, NoStacktrace, CancelledRequestException}
 import com.twitter.util._
 import java.util.concurrent.atomic.AtomicReference
 
+/**
+  * A base Dispatcher type to allow for easy tracing upon `WireRecv` and
+  * `WireSend` by the 
+  * [[com.twitter.finagle.transport.Transport transporter]]. Clients should
+  * provided custom behavior via the handle and dispatch methods
+  * @tparam Req the type of Request
+  * @tparam Rep the type of Response
+  * @tparam Out the type of message coming "out" of a
+  *  [[com.twitter.finagle.transport.Transport Transport]]
+  * @param init a [[ServerDispatcherInitializer]]
+  */
+abstract class ServerDispatcher[Req, Rep, Out](init: ServerDispatcherInitializer)
```

Comment:
this doesn't use ServerDispatcherInitializer, as far as I can tell.  let's not make it a constructor parameter until the layer where we need it.